app/src/test/java/org/mozilla/fenix/home/DefaultSessionControlControllerTest.kt

```diff
@@ -878,6 +881,61 @@ class DefaultSessionControlControllerTest {
         assertNull(TopSites.remove.testGetValue()!!.single().extra)
     }
 
+    @Test
+    fun `GIVEN exactly the required amount of downloaded thumbnails WHEN has no errors THEN handleShowWallpapersOnboardingDialog calls the dialog`() {
```

Comment:
nit: I think these tests might convey their purpose a bit better if they read like:

`GIVEN exactly the required amount of downloaded thumbnails with no errors WHEN handling wallpaper dialog THEN dialog is shown`

and 

`GIVEN fewer than the required amount of downloaded thumbnails WHEN handling wallpaper dialog THEN the dialog is not shown`